insights/parsers/uname.py

```diff
@@ -397,6 +397,24 @@ def _best_version(self, other):
         else:
             return self._lv_version, other._lv_version
 
+    def _lv_release_without_dist(self, other):
+        """
+        Get the LooseVersion releases without distribution part.
+        """
+        s_release_parts = self._lv_release.vstring.split(".")
+        o_release_parts = other._lv_release.vstring.split(".")
+        s_release = (
+            pad_release(".".join(s_release_parts[:-1]), len(s_release_parts))
+            if s_release_parts[-1].startswith("el")
+            else ".".join(s_release_parts)
+        )
+        o_release = (
+            pad_release(".".join(o_release_parts[:-1]), len(o_release_parts))
+            if o_release_parts[-1].startswith("el")
+            else ".".join(o_release_parts)
+        )
+        return LooseVersion(s_release), LooseVersion(o_release)
```

Comment:
Better `show` (not `raise`) a warning when only one of the operands contains the `distribution`. And when both of them contain `distribution`, we can take them into comparison. E.g.

```suggestion
        dist_opts = ('el',)
        s_release_parts = self._lv_release.vstring.split(".")
        o_release_parts = other._lv_release.vstring.split(".")
        s_dist = s_release_parts[-1]
        o_dist = o_release_parts[-1]
        s_release = ".".join(s_release_parts)
        o_release = ".".join(o_release_parts)
        if s_dist.startswith(dist_opts) ^ o_dist.startswith(dist_opts):
            import warnings
            warnings.warn('Comparison of distribution part will be ingored.')
            s_release = (
                s_release if not s_dist.startswith(dist_opts)
                else pad_release(".".join(s_release_parts[:-1]), len(s_release_parts))
            )
            o_release = (
                o_release if not o_dist.startswith(dist_opts)
                else pad_release(".".join(o_release_parts[:-1]), len(o_release_parts))
            )
        return LooseVersion(s_release), LooseVersion(o_release)
```

Reply:
Thanks, updated, please help take a look again.